docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md

```diff
@@ -1,166 +1,146 @@
 
-1. Click **Done** to finish creating the Okta application.
+   | Field                        | Description                                                                                                   |
+   | ---------------------------- | ------------------------------------------------------------------------------------------------------------- |
+   | `client_id`, `client_secret` | These values must match the client ID and client secret from your Okta OAuth2 app.                            |
+   | `auth_url`                   | The authorization endpoint of your OAuth2 provider. `https://<okta-tenant-id>.okta.com/oauth2v1authorize`     |
+   | `token_url`                  | The token endpoint of your Okta OAuth2 provider. `https://<okta-tenant-id>.okta.com/token`                    |
+   | `api_url`                    | The user information endpoint of your Okta OAuth2 provider. `https://<tenant-id>.okta.com/oauth2/v1/userinfo` |
+   | `enabled`                    | Enables Okta OAuth2 authentication. Set this value to `true`.                                                 |
 
-## Enable Okta OAuth in Grafana
+   Review the list of other Okta OAuth2 [configuration options]({{< relref "#configuration-options" >}}) and complete them, as necessary.
 
-1. Add the following to the [Grafana configuration file]({{< relref "../../../configure-grafana#configuration-file-location" >}}):
+1. Optional: [Configure a refresh token]({{< relref "#configure-a-refresh-token" >}}):
 
-```ini
-[auth.okta]
-name = Okta
-icon = okta
-enabled = true
-allow_sign_up = true
-client_id = some_id
-client_secret = some_secret
-scopes = openid profile email groups
-auth_url = https://<tenant-id>.okta.com/oauth2/v1/authorize
-token_url = https://<tenant-id>.okta.com/oauth2/v1/token
-api_url = https://<tenant-id>.okta.com/oauth2/v1/userinfo
-allowed_domains =
-allowed_groups =
-role_attribute_path =
-use_pkce = true
-```
+   a. Enable `accessTokenExpirationCheck` feature toggle.
 
-### PKCE
+   b. Extend the `scopes` field of `[auth.generic_oauth]` section in Grafana configuration file with refresh token scope used by your OAuth2 provider.
 
-IETF's [RFC 7636](https://datatracker.ietf.org/doc/html/rfc7636)
-introduces "proof key for code exchange" (PKCE) which provides
-additional protection against some forms of authorization code
-interception attacks. PKCE will be required in [OAuth 2.1](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-03).
+   c. Enable the refresh token on the provider if required.
 
-> You can disable PKCE in Grafana by setting `use_pkce` to `false` in the`[auth.okta]` section.
+1. [Configure role mapping]({{< relref "#configure-role-mapping" >}}).
+1. Optional: [Configure team synchronization]({{< relref "#configure-team-synchronization" >}}).
+1. Restart Grafana.
 
-### Configure refresh token
+   You should now see a generic OAuth2 login button on the login page and be able to log in or sign up with your OAuth2 provider.
```

Comment:
Here and elsewhere - do a quick search for "generic" and replace it with Okta 😄 

```suggestion
   You should now see an Okta login button on the login page and be able to log in or sign up with Okta.
```

Reply:
Done!